presto-main/src/main/java/io/prestosql/operator/scalar/MathFunctions.java

```diff
@@ -607,11 +607,47 @@ public static long random(@SqlType(StandardTypes.BIGINT) long value)
         return ThreadLocalRandom.current().nextLong(value);
     }
 
+    @Description("a pseudo-random number between m and n (exclusive")
+    @ScalarFunction(value = "random", alias = "rand", deterministic = false)
+    @SqlType(StandardTypes.TINYINT)
+    public static long rangeRandomTinyInt(@SqlType(StandardTypes.TINYINT) long initialValue, @SqlType(StandardTypes.TINYINT) long finalValue) {
+
+        checkCondition(initialValue <= finalValue, INVALID_FUNCTION_ARGUMENT, "origin must be less than bound");
+        return ThreadLocalRandom.current().nextInt((int) initialValue, (int) finalValue);
+    }
+
+    @Description("a pseudo-random number between m and n (exclusive")
+    @ScalarFunction(value = "random", alias = "rand", deterministic = false)
+    @SqlType(StandardTypes.SMALLINT)
+    public static long rangeRandomSmallInt(@SqlType(StandardTypes.SMALLINT) long initialValue, @SqlType(StandardTypes.SMALLINT) long finalValue) {
+
+        checkCondition(initialValue <= finalValue, INVALID_FUNCTION_ARGUMENT, "origin must be less than bound");
+        return ThreadLocalRandom.current().nextInt((int) initialValue, (int) finalValue);
+    }
+
+    @Description("a pseudo-random number between m and n (exclusive")
+    @ScalarFunction(value = "random", alias = "rand", deterministic = false)
+    @SqlType(StandardTypes.INTEGER)
+    public static long rangeRandomInteger(@SqlType(StandardTypes.INTEGER) long initialValue, @SqlType(StandardTypes.INTEGER) long finalValue) {
+
+        checkCondition(initialValue <= finalValue, INVALID_FUNCTION_ARGUMENT, "origin must be less than bound");
+        return ThreadLocalRandom.current().nextInt((int) initialValue, (int) finalValue);
+    }
+
+    @Description("a pseudo-random number between m and n (exclusive")
+    @ScalarFunction(alias = "rand", deterministic = false)
```

Comment:
name this function `value = "random"`, otherwise it's name will be `range_random`, which is not intentional